### ISSUE TITLE
create.barplot yaxis.lab does not work when NULL

### ISSUE THREAD

**Issue by raagagrawal** [uclahs-cds/public-R-BoutrosLab-plotting-general#62]
create.barplots' yaxis.lab option when set to NULL is meant to disable y axis tick labels.

However, it does not do that when yaxis.lab is set to NULL.

sample code:

``` r
library(BoutrosLab.plotting.general, quietly = T, warn.conflicts = F);
set.seed(10);
simple.data <- data.frame(
    x = rnorm(100),
    y = sample(1:5, replace = T, size = 100)
    );

# yaxis.lab = NULL removes y-axis tick labels in create.scatterplot
create.scatterplot(
    formula = y ~ x,
    data = simple.data,
    yaxis.lab = NULL
    );
```

![](https://i.imgur.com/kG5DFlO.png)

``` r

# does not remove in create.boxplot
create.boxplot(
    formula = y ~ x,
    data = simple.data,
    yaxis.lab = NULL
    );
```

![](https://i.imgur.com/ouJHmfb.png)

<sup>Created on 2023-01-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by stefaneng** [comment]
This seems to be a lattice issue unless there is some other way of removing the y-axis labels. The same underlying lattice code for `create.boxplot` is `lattice::bwplot`. To remove the yaxis labels for `lattice::xyplot` with lattice we can do: 
``` r
library(lattice);
set.seed(10);
simple.data <- data.frame(
    x = rnorm(100),
    y = sample(1:5, replace = T, size = 100)
    );
lattice::xyplot(y ~ x, data = simple.data, scales=list(y=list(labels=NULL)));
```

![](https://i.imgur.com/IuEw1oK.png)

``` r
# Does not work to remove y-axis labels
lattice::bwplot(y ~ x, data = simple.data, scales=list(y=list(labels=NULL)));
```

![](https://i.imgur.com/7lw2n49.png)

``` r
# Works fine to remove xaxis labels
lattice::bwplot(y ~ x, data = simple.data, scales=list(x=list(labels=NULL)));
```

![](https://i.imgur.com/0fnjbCs.png)

<sup>Created on 2023-01-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by stefaneng** [comment]
New info on how to fix this on the [lattice#26 issue](https://github.com/deepayan/lattice/issues/26)

> Well, `labels=NULL` is not documented to be a valid choice, so the behaviour is undefined. The reason for the inconsistency is that xyplot() and bwplot() have different "defaults" for `scales`.
> 
> My suggestion would be to either use
> 
> ```
> scales=list(y=list(at = numeric(0)))
> ```
> 
> (documented, but no tick marks), or
> 
> ```
> default.scales = list(y=list(labels = NULL))
> ```
> 
> (technically undefined, but likely to work for the foreseeable future).

If we define `axis.label = NULL` means removing the axis labels but keep the tick marks then we want to document this on all of the `create.*` functions and possibly the vignette as well.
